src/algebra/ordered_monoid.lean

```diff
@@ -49,22 +49,44 @@ class ordered_add_comm_monoid (α : Type*) extends add_comm_monoid α, partial_o
 
 attribute [to_additive] ordered_comm_monoid
 
+section ordered_instances
+
+@[to_additive]
+instance ordered_comm_monoid.to_covariant_class_left (M : Type*) [ordered_comm_monoid M] :
+  covariant_class M M (*) (≤) :=
+{ covc := λ a b c bc, ordered_comm_monoid.mul_le_mul_left _ _ bc a }
+
+@[to_additive]
+instance ordered_comm_monoid.to_covariant_class_right (M : Type*) [ordered_comm_monoid M] :
+  covariant_class M M (function.swap (*)) (≤) :=
+{ covc := λ a b c bc,
+    by { convert ordered_comm_monoid.mul_le_mul_left _ _ bc a; simp_rw mul_comm } }
+
+@[to_additive]
+instance ordered_comm_monoid.to_contravariant_class_left (M : Type*) [ordered_comm_monoid M] :
+  contravariant_class M M (*) (<) :=
+{ covtc := λ a b c bc, ordered_comm_monoid.lt_of_mul_lt_mul_left _ _ _ bc }
+
+@[to_additive]
+instance ordered_comm_monoid.to_contravariant_class_right (M : Type*) [ordered_comm_monoid M] :
+  contravariant_class M M (function.swap (*)) (<) :=
+{ covtc := λ a b c (bc : b * a < c * a), by { rw [mul_comm _ a, mul_comm _ a] at bc,
+    exact ordered_comm_monoid.lt_of_mul_lt_mul_left _ _ _ bc } }
+
+end ordered_instances
+
 /-- An `ordered_comm_monoid` with one-sided 'division' in the sense that
 if `a ≤ b`, there is some `c` for which `a * c = b`. This is a weaker version
 of the condition on canonical orderings defined by `canonically_ordered_monoid`. -/
 class has_exists_mul_of_le (α : Type u) [ordered_comm_monoid α] : Prop :=
 (exists_mul_of_le : ∀ {a b : α}, a ≤ b → ∃ (c : α), b = a * c)
 
-export has_exists_mul_of_le (exists_mul_of_le)
-
 /-- An `ordered_add_comm_monoid` with one-sided 'subtraction' in the sense that
 if `a ≤ b`, then there is some `c` for which `a + c = b`. This is a weaker version
 of the condition on canonical orderings defined by `canonically_ordered_add_monoid`. -/
 class has_exists_add_of_le (α : Type u) [ordered_add_comm_monoid α] : Prop :=
 (exists_add_of_le : ∀ {a b : α}, a ≤ b → ∃ (c : α), b = a + c)
 
-export has_exists_add_of_le (exists_add_of_le)
```

Comment:
Is there a reason to remove this (same with `mul`)?

Reply:
No real reason: I think that I removed it by mistake, while trying to figure out what instances and typeclasses to assume!  I will put it back!

Reply:
Then you can also remove a couple of `open has_exists_add_of_le` in various files.

Reply:
Ah, I see you did that.